### PR TITLE
Make IPC debug tracing configurable

### PR DESCRIPTION
**Pull request by alvsan09**
This change associates IPC tracing with the logger.debug method,
therefore enabling debugging enables IPC tracing.

Additionally, two new scripts are added to the top `package.json`, which can
be called as follows:

> yarn debug:electron
> yarn debug:browser

these enable starting of the electron and browser application examples in
debug mode.

Signed-off-by: Alvaro Sanchez-Leon <alvaro.sanchez-leon@ericsson.com>

#### What it does

Adds two new scripts to start the example applications (electron and browser) in debug mode,
it additionally associates IPC traces to debug mode.

#### How to test

1) Start the electron application in debug mode and verify that debug traces are present
> yarn debug:electron
2) Modify a file present in the workspace and check that IPC traces are displayed e.g.
image
3) close the theia example application
4) Start the electron application (no debugging mode), repeat step 2. and make sure no debug traces appear.
> yarn start:electron

Repeat the steps above replacing the `yarn` electron scripts for the corresponding `browser` scripts
i.e.
> yarn debug:browser
> yarn start:browser

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/node/messaging/ipc-connection-provider.ts [hunk 1]
```diff
@@ -81,11 +81,11 @@ export class IPCConnectionProvider {
             error: (message: string) => this.logger.error(`[${options.serverName}: ${childProcess.pid}] ${message}`),
             warn: (message: string) => this.logger.warn(`[${options.serverName}: ${childProcess.pid}] ${message}`),
             info: (message: string) => this.logger.info(`[${options.serverName}: ${childProcess.pid}] ${message}`),
-            log: (message: string) => this.logger.info(`[${options.serverName}: ${childProcess.pid}] ${message}`)
+            log: (message: string) => this.logger.debug(`[${options.serverName}: ${childProcess.pid}] ${message}`)
         });
-        connection.trace(Trace.Off, {
+        connection.trace(Trace.Verbose, {
```
Comment (vince-fugnitto) [inline packages/core/src/node/messaging/ipc-connection-provider.ts:86]: @alvsan09 I believe the idea would be to make this value configurable, not hardcoded to `off` or `verbose`.

**Review by vince-fugnitto — APPROVED** [review]
I confirmed that the tracing works at the `ipc` level (filesystem and git) with the changes.
- [x] `yarn start:electron`: no debug logs present.
- [x] `yarn debug:electron`: debug logs are present on the backend. 
- [x] `yarn start:browser`: no debug logs present.
- [x] `yarn debug:browser`: debug logs are present on the backend. 

An example log is the following:


```
root DEBUG [git-locator: 11545] Sending request 'locate - (0)'. Params: [
    "/Users/vincentfugnitto/workspace/theia",
    {}
]
```

I'll let others review as well to see if they have any feedback.

File: examples/electron/package.json [hunk 2]
```diff
@@ -63,7 +63,7 @@
     "bundle": "theia build --mode development",
     "watch": "concurrently -n compile,bundle \"theiaext watch --preserveWatchOutput\" \"theia build --watch --mode development\"",
     "start": "theia start --plugins=local-dir:../../plugins",
-    "start:debug": "yarn start --log-level=debug",
+    "start:debug": "yarn start --log-level=debug --remote-debugging-port=9222",
```
Comment (paul-marechal) [inline examples/electron/package.json:66]: This doesn't like a valid Node.js CLI argument: https://nodejs.org/docs/latest-v12.x/api/cli.html

Comment (paul-marechal) [reply examples/electron/package.json:66]: Ah looks like it's an Electron flag ok: https://www.electronjs.org/docs/api/command-line-switches#--remote-debugging-portport

Comment (alvsan09) [reply examples/electron/package.json:66]: right, this prepares the debug session so we can attach to it from the IDE

**Review by paul-marechal — APPROVED** [review]

**Pull request merged**